src/languages/gml.js

```diff
@@ -2994,135 +898,569 @@ export default function(hljs) {
+    ]
+  };
+
+  /**
+   * Match for the accessor for a `ds_map`. This exists to disambiguate from ternaries, which would
+   * otherwise consume the match for a struct literal member.
+   */
+  const DS_MAP_ACCESS = {
+    match: /\[\?/
+  };
+
+  /**
+   * A ternary expression, matching partial ternary as `? <EXPRESSION> :`.
+   * Effectively exists to prevent {@link STRUCT_LITERAL_MEMBER} from stealing `<EXPRESSION> :`.
+   */
+  const TERNARY = {
+    begin: /\?/,
+    end: /:/,
+    contains: EXPRESSION
+  };
+
+  const SWITCH_CASE = {
+    begin: [
+      /case/,
+      /\s+/
+    ],
+    end: /:/,
+    scope: {
+      1: "keyword"
+    },
+    contains: EXPRESSION
+  };
+  
+  /**
+   * A struct variable declaration, of `<ident>:`
+   */
+  const STRUCT_LITERAL_MEMBER = {
+    match: [
+      /\b/,
+      VALID_IDENTIFIER_RE,
+      /\s*:/
+    ],
+    scope: {
+      2: "variable"
+    },
+  };
+
+  /**
+   * A function declaration matching for:
+   * ```gml
+   * function <ident>(
+   * ```
+   */
+  const FUNCTION_DECLARATION = {
+    match: [
+      "function",
+      /\s+/,
+      VALID_IDENTIFIER_RE,
+      /\s*?\(/
+    ],
+    scope: {
+      1: "keyword",
+      3: "title.function"
+    }
+  };
+
+  /**
+   * An enum definition in the form:
+   * ```gml
+   * enum <ident> {
+   *     <ident> [= <expr>][,]
+   * }
+   * ```
+   */
+  const ENUM_DEFINITION = {
+    begin: [
+      /enum/,
+      /\s+/,
+      VALID_IDENTIFIER_RE,
+      /\s*{/
+    ],
+    end: "}",
+    scope: {
+      1: "keyword",
+      3: "variable.constant"
+    },
+    contains: [
+      COMMENT,
+      {
+        begin: [
+          VALID_IDENTIFIER_RE,
+          /\s*=\s*/
+        ],
+        end: /,|$|}/,
+        scope: {
+          1: "variable.constant"
+        },
+        contains: EXPRESSION
+      },
+      {
+        match: VALID_IDENTIFIER_RE,
+        scope: "variable.constant"
+      }
+    ]
+  };
+
+  EXPRESSION.push(
+    STRING,
+    DS_MAP_ACCESS,
+    TERNARY,
+    PROP_ACCESS,
+    NUMBER,
+    FUNCTION_CALL,
+    USER_ASSET_CONSTANT,
+    {
+      match: VALID_IDENTIFIER_RE,
+      keywords: {
+        keyword: KEYWORDS,
+        literal: LITERALS,
+        "variable.language": LANGUAGE_VARIABLES
+      },
+    },
+  );
+
+  /**
+   * Pre-processor modes for macro definitions and regions.
+   */
+  const PREPROCESSOR = {
+    end: /$/,
+    variants: [
+      {
+        begin: [
+          /#macro\s+/,
+          VALID_IDENTIFIER_RE
+        ],
+        end: /$/,
+        scope: {
+          1: "meta",
+          2: "variable.constant"
+        },
+        contains: [
+          ...EXPRESSION,
+          COMMENT,
+          {
+            match: /\\\n/
+          }
+        ]
+      },
+      {
+        begin: /#(end)?region\b/,
+        beginScope: "meta",
+        contains: [COMMENT_LINE_INNER]
+      },
+    ]
+  };
+
   return {
     name: 'GML',
-    case_insensitive: false, // language is case-insensitive
```

Comment:
Was this wrong?

Reply:
I removed this as the default value is already false. If it's preferable to define it explicitly, I'll put it back :)